pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -410,6 +412,12 @@ func CreateGCECloud(config *CloudConfig) (*GCECloud, error) {
 		kmsConfig:                config.KMSConfig,
 	}
 
+	// Register the provided KMS service with the plugin registry to allow using
+	// it for encryption of resources.
+	encryptionconfig.KMSPluginRegistry.Register(KMSServiceName, func(_ io.Reader) (envelope.Service, error) {
+		return gce.getGCPCloudKMSService()
```

Comment:
It is fetched once for every mention of that plugin in the encryption-provider-config file. That should be a very small number always.